### ISSUE TITLE
Filter functionality for Invoices/Orders lists

### ISSUE THREAD

**Issue by mariandl** [foodcoops/foodsoft#847]
Request by another person:

A function to filter invoices/order lists for a certain producers might increase use-, readability and thus productivity. Would be nice for:
- "finance>invoices" view ("finance" > "balancing")
- "orders" view ("orders" > "('manage'?) orders")

Thought on how/where this can be implemented:

- combobox
 
![orders-filter](https://user-images.githubusercontent.com/79068320/109956980-9cccfe80-7ce4-11eb-8bcc-82368b1b0257.jpg)

- search functionality
  - can be reused for other pages (implementing a search functionality might be better to discuss within another issue)
  - allows filtering for other properties/variables

![orders-filter-search](https://user-images.githubusercontent.com/79068320/109957283-0220ef80-7ce5-11eb-81f3-f8095bcb7530.jpg)

**Comment by mjavurek** [comment]
Thanks @mariandl for the nice visualisation!

**Comment by JohannHansing** [comment]
I played around with the code to try and enable sorting by the Supplier column via `sort_link_helper heading_helper(Order, :name), 'name'`. 

Strangely, the blue highlighting of the column header is enabled and clicking it produces the blue arrow but the ordering is not actually enforced (see attached screen shot). Am I missing the place where it's somehow hard coded that this table is always ordered by the End column (which is my impression)

<img width="1440" alt="Screen Shot 2021-11-06 at 14 48 47" src="https://user-images.githubusercontent.com/12748131/140612000-ec8b8ed5-c8c9-457d-adab-10b7fb8588b5.png">
?

**Comment by JuliusR** [comment]
> Am I missing the place where it's somehow hard coded that this table is always ordered by the End column (which is my impression)?

In the [`finished` scope](https://github.com/foodcoops/foodsoft/blob/6115cf58a10a88ef1873937abda05014f0d58afa/app/models/order.rb#L44), that ordering is enforced.

Should we override it with `.reorder`? I tried it in https://github.com/JuliusR/foodsoft/commit/2b14ea1da23d6f893cb3c0d775dc61642d4c351f, and it seems to work.

**Comment by JohannHansing** [comment]
>In the finished scope, that ordering is enforced.

Does anyone remember a specific reason why this is enforced? I personally don't see why it would be so if there is no strong argument for it that anyone is aware of I would suggest to let the users decide which column they want to order by.

**Comment by JuliusR** [comment]
> > In the finished scope, that ordering is enforced.
> 
> Does anyone remember a specific reason why this is enforced?

No, I do not know the reason in case of the `finished` scope. It does make sense for the `recent` scope:
https://github.com/foodcoops/foodsoft/blob/6115cf58a10a88ef1873937abda05014f0d58afa/app/models/order.rb#L33

I hesitate to touch the scopes because I cannot estimate the full consequences. However, we can `reorder` after the scope (as I tried in https://github.com/JuliusR/foodsoft/commit/28c837f7442dbde6558fd80c441a2da5087ebba8).
